curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-modern-javascript-methods-by-building-football-team-cards/63e94dae6dcedbad73f2f6ee.md

```diff
@@ -24,10 +24,13 @@ Inside the parameter list in the callback function for the `map` method, unpack
 You should unpack all 5 object properties from objects in `arr` using object destructuring.
 
 ```js
+const ARGS_BEFORE = "playerCards.innerHTML += arr.map((";
 
-assert.match(code, /{\s*((name|position|number|isCaptain|nickname)\s*,\s*(name|position|number|isCaptain|nickname)\s*,\s*(name|position|number|isCaptain|nickname)\s*,\s*(name|position|number|isCaptain|nickname)\s*,\s*(name|position|number|isCaptain|nickname))\s*}\s*/)
-
+let args = code.substring(code.indexOf(ARGS_BEFORE) + ARGS_BEFORE.length);
+args = args.substring(0, args.indexOf(")"));
+args = args.replaceAll(/\s/g, "");
 
+assert.equal(args, "{name,position,number,isCaptain,nickname}");
```

Comment:
there is an issue here, it's an object, the args can go in any order and still work, the regex as clunky as it was would allow that